### ISSUE TITLE
Making the site bilingual

### ISSUE THREAD

**Issue by alcarney** [CardiffMathematicsCodeClub/CardiffMathematicsCodeClub.github.io#224]
It's about time we catered for the Welsh speakers among us - plus maintaining translations etc. should be a nice entry level contribution for people.

The hard part will be setting this us up so that it's easy to keep the internals of both the English and Welsh versions in sync. Off the top of my head we have the following options:

- Probably the 'easiest' way to do this would be to simply copy the repository to a  CardiffMathematicsCodeClub....-cy repository and start translating and link between the two websites. But then this would put a huge burden on maintaining changes to the internals of the website. 

  We would need to either get people to open PRs against both sites and make sure they get merged in, or we would need somebody in charge of making sure changes get transferred between both sites. 

- There is a [plugin](http://jekyll-langs.liaohuqiu.net/) which makes the process quite easy. Say we had a blogpost `01-01-1970-welcome.md` we wanted translated to welsh then we put the translation in a file called `01-01-1970-welcome.cy.md`, set a few settings in `_config.yml` and we're away!

  Except this particular plugin isn't hosted on github pages so we would no longer be able to get the site built by Github. After each 'release' we would need someone to build the site on their machine and then push it to Github manually which introduces an extra layer of complexity. (That we don't really want #80) 

- Or we could roll our own solution, this would have the biggest upfront cost and we won't really know how much more complicated it could make the website until we have something that works. In the process of writing this issue I did find [this post](https://www.sylvaindurand.org/making-jekyll-multilingual/) which we might be able to use as a starting point. 

What are your thoughts?

**Comment by Huaraz2** [comment]
> I did find this post which we might be able to use as a starting point.

Am I uderstanding the post correctly, it seems to me as though we need a copy of every page in every language that we want to use, or am I missunderstanding it?

**Comment by alcarney** [comment]
@Huaraz2 No that's about right - which will be the case for whichever solution we go with

**Comment by Huaraz2** [comment]
> @Huaraz2 No that's about right - which will be the case for whichever solution we go with

Ok then this means that we will always need someone who is fluent in at least Welsh (as I asume this will be the first one we translate the site to) as a member of code club as I don't trust google translate that much, I haven't used it for along time so it may have improved the quality of its translation by now.

I sugest that we go with the solution sugested in [this post](https://www.sylvaindurand.org/making-jekyll-multilingual/) as it is the simplest idea as it means we can still use GitHub to host the site, and also that we translate the site to a number of different languages, not just welsh, as this could increase the interest in the site.

**Comment by alcarney** [comment]
> I don't trust google translate that much.

Same here, plus I feel it would be cheating anyway :p

> we could translate the site to a number of different languages

Great idea! I know we had a few French and Spanish speakers this year at
least - the more the merrier :D

**Comment by Huaraz2** [comment]
> I feel it would be cheating anyway :p

Meh not so muich as cheating more just taking a shortcut, which may or may not turn out to take longer.

**Comment by Huaraz2** [comment]
@alcarney is it possible to atomate github issue creation? ie if we have the site in English, Welsh and Spanish and I add a blogpost that is only in English, then sn issue is created for making a Welsh and Spanish version of my post?

**Comment by alcarney** [comment]
@Huaraz2 > to automate github issue creation

Yes 

But maybe not an issue for every post/page on the site. We could probably set up a monthly issue that tracks blog posts on a monthly basis in a checklist or something.

Then again we probably won't be getting > 5 news posts every month... So *shrug :p

**Comment by Huaraz2** [comment]
> Then again we probably won't be getting > 5 news posts every month...

If we make it into a socety we could do as the more people we have the more likely there will be something to write a post about

**Comment by Huaraz2** [comment]
Is there a way to set up a *live* conter oon the site that keeps track of the current no. of issues?

**Comment by alcarney** [comment]
@Huaraz2 I would think so

**Comment by geraintpalmer** [comment]
Great idea! My opinion:

> it seems to me as though we need a copy of every page in every language that we want to use

This seems like a waste of time. Most bilingual sites I seen make sure menus, titles, buttons etc are bilingual, but content intself (blog posts, news items) just in whatever language it was written in. So like it's be cool to have most things in English, but then now and again put a Welsh/French/etc post up about something specific to that language.

This saves on translators / relying on google translate (don't) too. Just my opinion.

**Comment by alcarney** [comment]
@geraintpalmer Perhaps things like blog posts we might not bother with translating. But would it be useful to translate say pages in the reference section?

**Comment by geraintpalmer** [comment]
> Perhaps things like blog posts we might not bother with translating. But would it be useful to translate say pages in the reference section?

Yeah good idea. And make the english page default (within the Welsh framework) if it hasn't been translated yet.

**Comment by Huaraz2** [comment]
@geraintpalmer :+1: to this idea
